### ISSUE TITLE
[FEATURE] Sort new Subscriptions alphabetically

### ISSUE THREAD

**Issue by ngInit** [Tzahi12345/YoutubeDL-Material#703]
**Is your feature request related to a problem? Please describe.**
I didn't see this mentioned anywhere else, but it would be nice if there was at least an option to re-sort your subscriptions in alphabetical order rather than order added.

**Describe the solution you'd like**
Subscriptions are sorted in alphabetical order by default, or provide an option to reorder your subscriptions to whatever order you'd like, either automatically or with some sort of drag and drop reordering.

**Additional context**
Currently any subscriptions that get added show up on the list (sidebar and subscriptions page) in whatever order it was added. This order doesn't change on a page refresh or container restart.

**Comment by Tzahi12345** [comment]
There's a couple solid improvements I can see here:

#### Subscriptions page can be a table

* Much more space efficient
* Solves your problem with sorting and all, at least in the subscriptions page (you can sort by header, use search, etc)
* You can add several buttons beyond info and editing - ex: "favorite" a subscription

#### Sidebar subscriptions don't need to be all subscriptions, they should be a subset of all subscriptions and sortable

"Favoriting" subscriptions such that they appear in the sidebar shouldn't be too difficult. The real issue is sorting. Why? Because we use a relational database. They show up in the order that they are added because that's how MongoDB/JSONs return the data back to us, but subscriptions don't have an inherent "order."

We could use a unique index to manage that, but we'd have to deal with the interoperability between the Local DB and MongoDB. With playlists, we avoid this problem by throwing away the "relational" aspect to the wayside -- playlists simply store a list of unique IDs for each video and we reorder *that.*

Things can get really complicated if we try to have it both ways. I'll propose two UI-side solutions.

1. Maintain a list of subscription IDs UI-side dictating the order of favorited subscriptions. Every time subscriptions are retrieved from the backend, this list is checked against the *actual* list of favorited subscriptions. Any subscriptions missing from the list (e.g. newly favorited subscriptions) just get added to the end, and any subscriptions *in* the UI-side list but don't actually exist (e.g. they were unsubscribed) get auto removed from the UI-side list.

2. Do the same as above, but store the list in the DB as a user setting. 

Benefit of option 2 is that the list is maintained on every device you use to access your instance. The downside is two-fold. First, we don't have a system to store per-user settings in the DB. Second, we'd have to make an extra database call every time you get your subscriptions, and every time the list is modified.

Perhaps there's a solution here I'm missing. Big picture though, how does it sound to have a sortable but not customizable list of subscriptions in a table in the subscriptions page, but have a reorderable list of "favorited" subscriptions in the sidebar?

**Comment by ngInit** [comment]
Yeah that could work. If you're trying to avoid database changes for things like the sidebar favorites, a solution where preferences/favorites are read from a json file in appdata folder might be simple enough. Something like that could also be useful for other features that you wouldn't necessarily want to modify the database for.
